### ISSUE TITLE
linter: `no-negated-condition` should (optionally) treat "is not falsy" as positive

### ISSUE THREAD

**Issue by aaronadamsCA** [oxc-project/oxc#22480]
I think a negative comparison to a [falsy value](https://developer.mozilla.org/en-US/docs/Glossary/Falsy), e.g. `x != null`, would be better treated as a positive condition. A double-negative is a positive, after all.

Today the rule suggests flipping to `x == null`, which IMO makes code harder to read:

```js
x ? use(x) : fallback(); // ⚠️ flagged by strict-boolean-expressions
x != null ? use(x) : fallback(); // ✅ safer, flagged by no-negated-condition
x == null ? fallback() : use(x); // ⚠️ less readable
```

Given #21569, I thought maybe this could be a new option for an existing Oxlint rule, maybe `unicorn/no-negated-condition`. Filing upstream with ESLint or Unicorn seems pointless; an incompatible behaviour change seems iffy; a new option felt like a good compromise.

Examples of **incorrect** code with this new option:

```js
a == null ? null : <Profile user={a} />;
x === undefined ? fallback() : use(x);
flag === false ? bail() : proceed(flag);
Number.isNaN(n) ? <Empty /> : render(n);

if (count === 0) {
  return <Empty />;
} else {
  return render(count);
}
```

Examples of **correct** code with this new option:

```js
a != null ? <Profile user={a} /> : null;
x !== undefined ? use(x) : fallback();
flag !== false ? proceed(flag) : bail();
!Number.isNaN(n) ? render(n) : <Empty />;

if (count !== 0) {
  return render(count);
} else {
  return <Empty />;
}
```

**Comment by camc314** [comment]
Hi! Thanks for suggesting this change.

Changing this is not planned at this time.

I would recommend writing a custom JS plugin to achieve the desired behaviour.

Thanks!
